src/pilot/assign_role.py

```diff
@@ -781,7 +785,12 @@ def configure_raid(ironic_client, node_uuid, role, os_volume_size_gb,
         role, drac_client)
 
     if target_raid_config is None:
-        return False
+        return None
+
+    if "logical_disks" in target_raid_config and not\
```

Comment:
Looking at define_target_raid_config, it will either return None or a dictionary with a 'logical_disks' key.  As a result, I think you can simplify the above line to:
if not target_raid_config['logical_disks']:

Reply:
To be on safe side i checked `logical_disks` key present or not in target_raid_config. But now as we split the condition to check None in target_raid_config, i think `if not target_raid_config['logical_disks']:` can works. so removed as you suggested.